src/django/api/views.py

```diff
@@ -1280,3 +1288,193 @@ def api_feature_flags(request):
+    permission_classes = [IsAdminUser]
+
+    def create(self, request):
+        pass
+
+    def delete(self, request):
+        pass
+
+    def list(self, request):
+        if not switch_is_active('claim_a_facility'):
+            raise NotFound()
+
+        if not request.user.is_superuser:
+            raise PermissionDenied()
+
+        response_data = FacilityClaimSerializer(FacilityClaim.objects.all(),
+                                                many=True).data
+
+        return Response(response_data)
+
+    def retrieve(self, request, pk=None):
+        if not switch_is_active('claim_a_facility'):
+            raise NotFound()
+
+        if not request.user.is_superuser:
+            raise PermissionDenied()
+
+        try:
+            claim = FacilityClaim.objects.get(pk=pk)
+            response_data = FacilityClaimDetailsSerializer(claim).data
+
+            return Response(response_data)
+        except FacilityClaim.DoesNotExist:
+            raise NotFound()
+
+    @transaction.atomic
+    @action(detail=True,
+            methods=['post'],
+            url_path='approve')
+    def approve_claim(self, request, pk=None):
+        if not switch_is_active('claim_a_facility'):
+            raise NotFound()
+
+        if not request.user.is_superuser:
+            raise PermissionDenied()
+
+        try:
+            claim = FacilityClaim.objects.get(pk=pk)
+
+            if claim.status != FacilityClaim.PENDING:
+                raise APIException('Only PENDING claims can be approved.')
+
+            claim.approval_reason = request.data.get('reason', '')
+            claim.approved_by = request.user
+            claim.approval_date = datetime.now()
+            claim.status = FacilityClaim.APPROVED
+            claim.save()
+
+            note = 'Status was updated to {} for reason: {}'.format(
+                FacilityClaim.APPROVED,
+                claim.approval_reason,
+            )
+
+            FacilityClaimReviewNote.objects.create(
+                claim=claim,
+                author=request.user,
+                note=note,
+            )
+
+            send_claim_facility_approval_email(request, claim)
+
+            response_data = FacilityClaimDetailsSerializer(claim).data
+            return Response(response_data)
+        except FacilityClaim.DoesNotExist:
+            raise NotFound()
+
+    @transaction.atomic
+    @action(detail=True,
+            methods=['post'],
+            url_path='deny')
+    def deny_claim(self, request, pk=None):
+        if not switch_is_active('claim_a_facility'):
+            raise NotFound()
+
+        if not request.user.is_superuser:
+            raise PermissionDenied()
+
+        try:
+            claim = FacilityClaim.objects.get(pk=pk)
+
+            if claim.status != FacilityClaim.PENDING:
+                raise APIException('Only PENDING claims can be denied.')
+
+            claim.denial_reason = request.data.get('reason', '')
```

Comment:
I believe these need to be replaced by `status_change_*` properties.

Reply:
Fixed up in 204d9548f31cde4d35f74618dc355f6746b7288a